bin/pycbc_upload_xml_to_gracedb

```diff
@@ -181,70 +192,91 @@ for event in coinc_table:
-        # and move to the next event
         logging.info("Not uploading event")
-        xmldoc.childNodes[-1].removeChild(coinc_event_table_curr)
-        xmldoc.childNodes[-1].removeChild(coinc_inspiral_table_curr)
-        xmldoc.childNodes[-1].removeChild(coinc_event_map_table_curr)
-        xmldoc.childNodes[-1].removeChild(sngl_inspiral_table_curr)
-        continue
-
-    group_tag = 'Test' if args.testing else 'CBC'
-    r = gracedb.createEvent(group_tag, 'pycbc', filename_xml,
-                            filecontents=open(fullpath_xml, "rb").read(),
-                            search=id_str, offline=True).json()
-
-    logging.info("Uploaded event %s.", r["graceid"])
-
-    # upload PSD
-    gracedb.writeLog(
-        r["graceid"], "PyCBC PSD estimate from the time of event",
-        "psd.xml.gz", open(fullpath_psd, "rb").read(), "psd").json()
-    logging.info("Uploaded file %s to event %s.", filename_psd, r["graceid"])
-
-    # add info for tracking code version
-    gracedb_tag_with_version(gracedb, r['graceid'])
-
-    # document the absolute path to the input file
-    input_file_str = 'Candidate uploaded from ' \
-        + os.path.abspath(args.input_file)
-    gracedb.writeLog(r['graceid'], input_file_str)
-
-    # add the custom log message, if provided
-    if args.log_message is not None:
-        gracedb.writeLog(r['graceid'], args.log_message,
-                         tag_name=['analyst_comments'])
+    else:
+        group_tag = 'Test' if args.testing else 'CBC'
+        r = gracedb.createEvent(group_tag, 'pycbc', filename_xml,
+                                filecontents=open(fullpath_xml, "rb").read(),
+                                search=id_str, offline=True).json()
+
+        logging.info("Uploaded event %s.", r["graceid"])
+
+        # add info for tracking code version
+        gracedb_tag_with_version(gracedb, r['graceid'])
+
+        # document the absolute path to the input file
+        input_file_str = 'Candidate uploaded from ' \
+            + os.path.abspath(args.input_file)
+        gracedb.writeLog(r['graceid'], input_file_str)
+
+        # add the custom log message, if provided
+        if args.log_message is not None:
+            gracedb.writeLog(r['graceid'], args.log_message,
+                             tag_name=['analyst_comments'])
+
+    if args.generate_plots:
+        psd_fig, psd_ax = plt.subplots(1)
+        for ifo in sorted(psddict.keys()):
+            curr_psd = psddict[ifo]
+            psd_ax.loglog(curr_psd.sample_frequencies,
+                          curr_psd ** 0.5 / pycbc.DYN_RANGE_FAC,
+                          c=ifo_color(ifo), label=ifo)
+        psd_ax.legend()
+        psd_ax.set_xlim([10, 1300])
+        psd_ax.set_ylim([3E-24, 1E-20])
+        psd_ax.set_xlabel('Frequency (Hz)')
+        psd_ax.set_ylabel('ASD')
+        psd_png_filename = '{}-PYCBC_{}-{:d}-psd.png'.format(ifos_str, id_str,
+                                                             time)
```

Comment:
```suggestion
        psd_png_filename = f'{ifos_str}-PYCBC_f{id_str}_PSD-{time:d}-1.png'
```
Note that I made the file name consistent with the usual convention (duration at the end, even if bogus) and moved the "PSD" label into the arbitrary-string part. Maybe you did it the other way because you want the file names to naturally sort lexicographycally and group event files and their PSD plots together?

Reply:
Looking at it now, we plot the ASD, not the PSD, so maybe that should be updated in the naming for both plots?

Reply:
Good point. Some of us may be used to just calling it "PSD" anyway, and would be confused by not finding a file with "PSD" in the name, but this does not seem a good reason to use the wrong terminology.